webapp/tests/karma/ts/services/enketo.service.spec.ts

```diff
@@ -1542,4 +1542,102 @@ describe('Enketo service', () => {
       expect(service.setFormTitle.args[0][1]).to.be.equal('translated sentence New Area');
     });
   });
+
+  describe('multimedia', () => {
+    let overrideNavigationButtonsStub;
+    let pauseStubs;
+    let form;
+    let $form;
+    let $nextBtn;
+    let $prevBtn;
+    let originalJQueryFind;
+
+    before(() => {
+      $nextBtn = $('<button class="btn next-page"></button>');
+      $prevBtn = $('<button class="btn previous-page"></button>');
+      originalJQueryFind = $.fn.find;
+      overrideNavigationButtonsStub = sinon
+        .stub(EnketoService.prototype, <any>'overrideNavigationButtons')
+        .callThrough();
+
+      form = {
+        calc: { update: sinon.stub() },
+        output: { update: sinon.stub() },
+        pages: {
+          _next: sinon.stub(),
+          _getCurrentIndex: sinon.stub()
+        }
+      };
+    });
+
+    beforeEach(() => {
+      $form = $(`<div></div>`);
+      $form
+        .append($nextBtn)
+        .append($prevBtn);
+
+      pauseStubs = [];
+      sinon
+        .stub($.fn, 'find')
+        .callsFake(selector => {
+          const result = originalJQueryFind.call($form, selector);
+
+          result.each((idx, element) => {
+            if (element.pause) {
+              pauseStubs.push(sinon.stub(element, 'pause'));
+            }
+          });
```

Comment:
Can you just have these be stubbed up here at the top when the elements are created? I might be missing something here, so please let me know if I am. Otherwise it is preferred that they are created as stubs so the assertions work as one might expect. The dynamic stubbing in the `callsFake` seems to result in unusual assertions later, distracting from what is actually under test. If they are always stubbed, down when you do your check it will be much more readable, like this:

```
expect(nextPause.callCount).to.equal(1);
```

I'm also a fan of dsl methods like [once](https://sinonjs.org/releases/latest/mocks/#expectationonce) to be clearer of what's going on and avoid a reader having to consider what equality check might be happening. That's a smaller thing though and not how this is done above in the file, so I'd rather refactor it all (not today!) than create a one-off case here.

Reply:
> Can you just have these be stubbed up here at the top when the elements are created? 

I tried to stub the `pause` function when creating the objects, but then the stubs are missing when `jquery.find` returns the elements. So that's why I had to stub right when jquery find the elements at this point.

To improve readability, I can push the stubs into a map by using the element tag as key

```js
// When stubbing
if (element.pause) {
  pauseStubs[element.tagname] = sinon.stub(element, 'pause');
}

// when asserting 
expect(pauseStubs.VIDEO.callCount)......
```

> I'm also a fan of dsl methods like once

Nice! I'm going to try this out!

Reply:
Done